activerecord/test/cases/arel/nodes/filter_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require_relative "../helper"
+
+module Arel
+  module Nodes
+    class ::FilterTest < Arel::Spec
+      describe "Filter" do
+        it "should add filter to expression" do
+          table = Arel::Table.new :users
+          _(table[:id].count.filter(table[:gdp_per_capita].gteq(40_000)).to_sql).must_be_like %{
+              COUNT("users"."id") FILTER (WHERE "users"."gdp_per_capita" >= 40000)
+            }
```

Comment:
A small nitpick here is that to me GDP per capita doesn’t make much sense on users as an example. Perhaps using `income`, instead?

```suggestion
          _(table[:id].count.filter(table[:income].gteq(40_000)).to_sql).must_be_like %{
              COUNT("users"."id") FILTER (WHERE "users"."income" >= 40000)
            }
```